docs/amber.md

```diff
@@ -0,0 +1,138 @@
+                                        | Script |---------------->| Requirements |
+                                        +--------+                 +--------------+
+                                             |
+             +------------------+------------+--------+----------------------+
+             |                  |                     |                      |
+             v                  v                     v                      v
+      +---------+          +---------+        +---------+            +---------+
+      |  +--------+        |  +---------+     |  +----------+        |  +---------+
+      +--|  +--------+     +--|  +--------+   +--|  +----------+     +--|  +---------+
+         +--| Shader |        +--| Buffer |      +--| Pipeline |        +--| Command |
+            +--------+           +--------+         +----------+           +---------+
+               ^                  ^  ^                 |  |  ^               |    |
+               |                  |  |                 |  |  +---------------+    |
+Entry point    |                  |  +-----------------+  |                       |
+Optimizations  |                  |  Descriptor Set       |                       |
+Type           |                  |  Binding              |                       |
+Compiled Shader|                  |  Attachment Location  |                       |
+               |                  |                       |                       |
+               +------------------------------------------+                       |
+                                  |                                               |
+                                  +-----------------------------------------------+
+                                                        BufferCommand
+```
+
+A `Script` is made up for shaders, pipelines, buffers and commands. Pipelines
+contain shaders and buffers. The buffers hold both image and storage buffers.
+It is assumed that the `Buffer` object has the current value for that buffer at
+any given time. This means that a draw command will need to copy buffer data to
+the device, execute the draw, and then copy the device data back into the
+buffer. Amber does not do any extra calls to fill the buffers. Then engine must
+keep that data in sync.
+
+The `Pipeline` object holds the context for a given set of tests. This includes
+shaders, colour attachments, depth/stencil attachment, data buffers, etc. The
+colour attachments will have their `Attachment Location` while data buffers
+will have a `Descriptor Set` and `Binding` provided.
+
+## Execution
+When the script is executed the pipeline shaders will be compiled, if not
+provided through the shader map. This will fill in the `Compiled Shader` data
+in the each pipeline shader object. The `CreatePipeline` call will then be
+executed for a given pipeline.
+
+With the pipeline created on the backed each of the `Command` objects will
```

Comment:
Grammar?  I can't parse this, especially around "the backed"

Reply:
done.